src/components/App.tsx

```diff
@@ -6037,7 +6037,9 @@ class App extends React.Component<AppProps, AppState> {
 
           this.setState((prevState) => {
             const nextSelectedElementIds = {
-              ...(shouldReuseSelection && pointerDownState.selectedElementIds),
+              ...(shouldReuseSelection &&
+                event.shiftKey &&
```

Comment:
Originally I was going to modify `shouldReuseSelection`, but I was afraid it would affect the others so I just added the judgment here.
Any suggestions are welcome!